### ISSUE TITLE
Book improvements v2

### ISSUE THREAD

**Issue by Rhuagh** [slide-rs/specs#408]
- [x] #327 - Incorporate diagrams from @torkleyy's computer
- [x] Add `saveload` chapter 
- [x] Update chapter about tracked storage

**Comment by aspurdy** [comment]
I have a small suggestion for chapter 3 of the book. It might be more instructive to have the HelloWorld system depend on UpdatePos, so that the updated values are printed rather than the initial ones:

```rust
let mut dispatcher = DispatcherBuilder::new()
    .with(UpdatePos, "update_pos", &[])
    .with(HelloWorld, "hello_world", &["update_pos"])                 
    .build();                                                         
```

**Comment by torkleyy** [comment]
@aspurdy Sounds great, maybe we could even add `HelloWorld` twice, one before the update and one after. Would you like to create a PR for that?

**Comment by aspurdy** [comment]
@torkleyy Sure thing, I'd be happy to contribute that.

**Comment by ldesgoui** [comment]
Hello,

Should I wait on #460 and #461 being finished before working on adding the `saveload` chapter ?

Thanks.

**Comment by torkleyy** [comment]
Hey, sorry for the late reply. I think that would make sense, yes. One potential change might be a combination of two traits (`FromDeserialize` and `ToSerialize`), so that would affect the book.

**Comment by jrmiller82** [comment]
Would be super nice to have that `saveload` chapter. :)

**Comment by ldesgoui** [comment]
Hello,

I forgot about this issue, apologies.
I haven't began any work yet, if you'd like to tackle this, please feel free.
If not, I'll try dedicating some time in the following weeks.

Thanks

**Comment by torkleyy** [comment]
That would be great, thank you @ldesgoui!

**Comment by torkleyy** [comment]
Fixed by #589! Thanks @ldesgoui!
